public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -0,0 +1,321 @@
+// Libraries
+import React, { ReactNode } from 'react';
+import intersection from 'lodash/intersection';
+
+import {
+  QueryField,
+  SlatePrism,
+  FormField,
+  TypeaheadInput,
+  TypeaheadOutput,
+  BracesPlugin,
+  Select,
+  MultiSelect,
+} from '@grafana/ui';
+
+// Utils & Services
+// dom also includes Element polyfills
+import { Plugin, Node } from 'slate';
+import syntax from '../syntax';
+
+// Types
+import { ExploreQueryFieldProps, AbsoluteTimeRange, SelectableValue } from '@grafana/data';
+import { CloudWatchQuery, CloudWatchLogsQuery } from '../types';
+import { CloudWatchDatasource } from '../datasource';
+import Prism, { Grammar } from 'prismjs';
+import { CloudWatchLanguageProvider } from '../language_provider';
+import { css } from 'emotion';
+
+export interface CloudWatchLogsQueryFieldProps extends ExploreQueryFieldProps<CloudWatchDatasource, CloudWatchQuery> {
+  absoluteRange: AbsoluteTimeRange;
+  onLabelsRefresh?: () => void;
+  ExtraFieldElement?: ReactNode;
+  syntaxLoaded: boolean;
+  syntax: Grammar;
+}
+
+const containerClass = css`
+  flex-grow: 1;
+  min-height: 35px;
+`;
+
+const rowGap = css`
+  gap: 3px;
+`;
+
+interface State {
+  selectedLogGroups: Array<SelectableValue<string>>;
+  availableLogGroups: Array<SelectableValue<string>>;
+  loadingLogGroups: boolean;
+  regions: Array<SelectableValue<string>>;
+  selectedRegion: SelectableValue<string>;
+}
+
+export class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogsQueryFieldProps, State> {
+  state: State = {
+    selectedLogGroups:
+      (this.props.query as CloudWatchLogsQuery).logGroupNames?.map(logGroup => ({
+        value: logGroup,
+        label: logGroup,
+      })) ?? [],
+    availableLogGroups: [],
+    regions: [],
+    selectedRegion: (this.props.query as CloudWatchLogsQuery).region
+      ? {
+          label: (this.props.query as CloudWatchLogsQuery).region,
+          value: (this.props.query as CloudWatchLogsQuery).region,
+          text: (this.props.query as CloudWatchLogsQuery).region,
+        }
+      : { label: 'default', value: 'default', text: 'default' },
+    loadingLogGroups: false,
+  };
+
+  plugins: Plugin[];
+
+  constructor(props: CloudWatchLogsQueryFieldProps, context: React.Context<any>) {
+    super(props, context);
+
+    Prism.languages['cloudwatch'] = syntax;
+    this.plugins = [
+      BracesPlugin(),
+      SlatePrism({
+        onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
+        getSyntax: (node: Node) => 'cloudwatch',
+      }),
+    ];
+  }
+
+  fetchLogGroupOptions = async (region: string) => {
+    try {
+      const logGroups: string[] = await this.props.datasource.describeLogGroups({
+        refId: this.props.query.refId,
+        region,
+      });
+
+      return logGroups.map(logGroup => ({
+        value: logGroup,
+        label: logGroup,
+      }));
+    } catch (err) {
+      console.error(err);
```

Comment:
I think it would be better to emit alertEvent so we show user the error if there is some.

Reply:
👍